internal/transform/span.go

```diff
@@ -85,14 +85,32 @@ func (m *mapper) Transform(invocationStartedTimestamp int64) telemetry.Span {
 			Root: awsRoot,
 		},
 	}
-
+	lambdaName := os.Getenv("AWS_LAMBDA_FUNCTION_NAME")
 	lambdaType := "function"
-	if m.span.Name() != lumigoSpan.LambdaName && m.span.Name() != "LumigoParentSpan" {
+	if m.span.Name() != lambdaName && m.span.Name() != "LumigoParentSpan" {
 		lambdaType = "http"
 		lumigoSpan.SpanInfo.HttpInfo = m.getHTTPInfo(attrs)
+	} else {
+		lumigoSpan.LambdaName = lambdaName
+		lumigoSpan.MemoryAllocated = os.Getenv("AWS_LAMBDA_FUNCTION_MEMORY_SIZE")
+		lumigoSpan.Runtime = os.Getenv("AWS_EXECUTION_ENV")
+		if event, ok := attrs["event"]; ok {
+			lumigoSpan.Event = fmt.Sprint(event)
+		} else {
+			m.logger.Error("unable to fetch event")
+		}
+
+		isWarmStart := os.Getenv("IS_WARM_START")
+		if isWarmStart == "" && !isProvisionConcurrencyInitialization() {
+			lumigoSpan.LambdaReadiness = "cold"
+		} else {
+			lumigoSpan.LambdaReadiness = "warm"
+		}
 	}
 	lumigoSpan.LambdaType = lambdaType
-
+	if isStartSpan {
+		lumigoSpan.LambdaEnvVars = m.getEnvVars()
```

Comment:
We send the env vars on both start-span and end-span, all the data from the start-span should also be on the end-span.
So maybe it should be `if isFunctionSpan`